src/main/java/org/craftercms/studio/impl/v2/service/content/internal/ContentTypeServiceInternalImpl.java

```diff
@@ -75,36 +89,113 @@ public class ContentTypeServiceInternalImpl implements ContentTypeServiceInterna
+    protected final Cache<String, ContentTypeConfigTO> cache;
+
+    @ConstructorProperties({"securityService", "configurationService", "itemDao",
+            "siteService", "studioConfiguration", "contentRepository", "contentTypeBasePathPattern", "contentTypeDefinitionFilename",
+            "templateXPath", "controllerPattern", "controllerFormat", "previewImageXPath", "defaultPreviewImagePath", "cache"})
+    public ContentTypeServiceInternalImpl(SecurityService securityService,
                                           ConfigurationService configurationService, ItemDAO itemDao,
-                                          SiteService siteService, String contentTypeBasePathPattern,
+                                          SiteService siteService, StudioConfiguration studioConfiguration,
+                                          ContentRepository contentRepository, String contentTypeBasePathPattern,
                                           String contentTypeDefinitionFilename, String templateXPath,
                                           String controllerPattern, String controllerFormat,
-                                          String previewImageXPath, String defaultPreviewImagePath) {
-        this.contentTypeService = contentTypeService;
+                                          String previewImageXPath, String defaultPreviewImagePath, Cache<String, ContentTypeConfigTO> cache) {
         this.securityService = securityService;
         this.configurationService = configurationService;
         this.itemDao = itemDao;
         this.siteService = siteService;
+        this.studioConfiguration = studioConfiguration;
+        this.contentRepository = contentRepository;
         this.contentTypeBasePathPattern = contentTypeBasePathPattern;
         this.contentTypeDefinitionFilename = contentTypeDefinitionFilename;
         this.templateXPath = templateXPath;
         this.controllerPattern = controllerPattern;
         this.controllerFormat = controllerFormat;
         this.previewImageXPath = previewImageXPath;
         this.defaultPreviewImagePath = defaultPreviewImagePath;
+        this.cache = cache;
     }
 
     public void setContentService(ContentService contentService) {
         this.contentService = contentService;
     }
 
     @Override
-    public List<QuickCreateItem> getQuickCreatableContentTypes(String siteId) {
-        return contentTypeService.getAllContentTypes(siteId, true).stream()
+    public ContentTypeConfigTO loadContentTypeConfiguration(String siteId, String contentTypeId) {
+        String siteConfigPath = getConfigPath().replaceAll(PATTERN_SITE, siteId)
+                .replaceAll(PATTERN_CONTENT_TYPE, contentTypeId)
+                .replaceAll(FILE_SEPARATOR + FILE_SEPARATOR, FILE_SEPARATOR);
+        String configFileFullPath = siteConfigPath + FILE_SEPARATOR + getConfigFileName();
+
+        // TODO: SJ: Add general lock service lock around this key to avoid having more than one thread do this work
```

Comment:
Please let's take care of this TODOs (or let's discuss if still relevant) when porting code from v1